### DISCUSSION TITLE
Add support for authentication with IndieAuth

### DISCUSSION THREAD

**Discussion by capjamesg** [bluesky-social/atproto#1155] [Protocol (atproto)]
Bluesky / AT Protocol implementations are well-positioned to allow authentication with [IndieAuth](https://indieweb.org/IndieAuth), a standard that lets you sign into a service using your domain name. IndieAuth is a mature W3C recommendation now maintained by the IndieWeb community. There are many [implementations](https://indieweb.org/IndieAuth#Server_Implementations) and [consumers](https://indieweb.org/IndieAuth#IndieAuth_Clients). Because AT Protocol allows domains as usernames, it feels prudent to allow people to authenticate with those names, too.

IndieAuth works well with [RelMeAuth](https://microformats.org/wiki/RelMeAuth) where a site advertises their profiles on services and a server uses the rel=me links to decide what authentication methods to offer to the IndieAuth server. For instance, using IndieAuth I can sign into services with my email address because I have it on my website. I could also sign in with GitHub if I put a rel=me link to my GitHub on my home page. This applies to other services, too (depending on what authentication methods are supported by the IndieAuth server).

**Comment by aaronpk** [comment]
👋  IndieAuth and OAuth spec editor here! Happy to help figure this out for ATProto and Bluesky.

I was originally going to write some thoughts in this GitHub issue, but it quickly turned into a very long blog post. So I posted it on my blog instead!

https://aaronparecki.com/2023/03/09/5/bluesky-and-oauth
